cmd/serve/httplib/serve/data/data.go

```diff
@@ -0,0 +1,40 @@
+//go:generate go run assets_generate.go
+// The "go:generate" directive compiles static assets by running assets_generate.go
+
+package data
+
+import (
+	"html/template"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+)
+
+// GetTemplate eturns the HTML template for serving directories via HTTP
+func GetTemplate() (tpl *template.Template, e error) {
+	tpl = nil
+	e = nil
+
+	templateFile, err := Assets.Open("index.html")
+	if err != nil {
+		e = errors.New("Failed to open template")
```

Comment:
Idiomatic go would be to use `errors.Wrap` to add some context to the error, like this

```go
return nil, errors.Wrap(err, "get template open")
```